src/AutoRest.CSharp/Mgmt/Decorator/ReferenceTypePropertyChooser.cs

```diff
@@ -33,48 +37,56 @@ private static IList<System.Type> GetReferenceClassCollection()
             return assembly.GetTypes().Where(t => t.GetCustomAttributes(false).Where(a => a.GetType().Name == PropertyReferenceAttributeName).Count() > 0).ToList();
         }
 
-        public static ObjectTypeProperty? GetExactMatch(ObjectTypeProperty originalType, MgmtObjectType typeToReplace, ObjectTypeProperty[] properties)
+        public static CSharpType? GetExactMatch(MgmtObjectType typeToReplace)
         {
+            if (_valueCache.TryGetValue(typeToReplace, out var result))
+                return result;
             foreach (System.Type replacementType in GetReferenceClassCollection())
             {
-                // flatten properties
-                List<PropertyInfo> replacementTypeProperties = replacementType.GetProperties(BindingFlags.Public | BindingFlags.Instance).ToList();
-                List<PropertyInfo> flattenedReplacementTypeProperties = new List<PropertyInfo>();
-                foreach (var parentProperty in replacementTypeProperties)
+                // Quick match for models in Resources SDK with models in the same namespace of Core package through the full name of "Namespace.Name"
+                if ($"{typeToReplace.Type.Namespace}.{typeToReplace.Type.Name}" == $"{replacementType.Namespace}.{replacementType.Name}")
```

Comment:
As the ErrorResponse used by Resources SDK and the ErrorResponse in common-type v2 of the Azure.REsourceManager package will be 2 different models, will dump this quick match logic for Resources SDK.